browser/component_updater/brave_component_contents_verifier.cc

```diff
@@ -0,0 +1,193 @@
+    0xd8, 0x88, 0xc2, 0x11, 0x7e, 0xf7, 0x5d, 0xfc, 0xd6, 0x99, 0x50, 0x9b,
+    0x70, 0xe9, 0xa7, 0xcf, 0x7f, 0x4c, 0xf0, 0x80, 0x2d, 0x8d, 0x97, 0x9f,
+    0xfc, 0x5c, 0x56, 0xc3, 0xff, 0x37, 0xf8, 0x3a, 0x44, 0x3a, 0x44, 0x3c,
+    0xbe, 0x96, 0xf9, 0x14, 0x44, 0xb1, 0x8e, 0x08, 0xc2, 0x55, 0x49, 0xf1,
+    0xe4, 0x3a, 0xa6, 0x63, 0x23, 0x26, 0x00, 0xa4, 0x8b, 0xdd, 0xaf, 0x45,
+    0xb0, 0xf8, 0xb8, 0x24, 0x80, 0x85, 0x85, 0x95, 0x2e, 0xc5, 0xd4, 0xf1,
+    0xdc, 0xfb, 0xeb, 0x95, 0xb6, 0x13, 0x3a, 0x46, 0x9d, 0x49, 0xb1, 0x15,
+    0x72, 0xfd, 0x85, 0xc6, 0x0b, 0x7e, 0xe2, 0x97, 0xf1, 0x66, 0xa6, 0x0b,
+    0x67, 0xe8, 0x72, 0xb9, 0xf1, 0x3b, 0x30, 0x1a, 0x77, 0x6d, 0xf3, 0xc4,
+    0x99, 0xc0, 0x86, 0xb2, 0x4e, 0x1d, 0xde, 0x34, 0x86, 0x15, 0x41, 0x68,
+    0x1e, 0x94, 0x48, 0x0a, 0x54, 0x05, 0x52, 0x01, 0x04, 0xd2, 0xef, 0x57,
+    0x89, 0x47, 0xd3, 0xae, 0xd9, 0xc1, 0xf5, 0xa9, 0xbf, 0x60, 0x96, 0x4b,
+    0xa0, 0x12, 0x9c, 0xf3, 0x00, 0xe6, 0x32, 0x40, 0xc7, 0x6e, 0x29, 0xa8,
+    0x81, 0xfd, 0x2f, 0x1e, 0x92, 0x4a, 0x5e, 0xed, 0xef, 0x13, 0x9f, 0xed,
+    0x88, 0x77, 0x2c, 0x84, 0xdd, 0x00, 0x87, 0x03, 0x49, 0x09, 0xb7, 0x4b,
+    0xc7, 0x02, 0x03, 0x01, 0x00, 0x01};
+
+constexpr const char kVerifiedContentsPath[] =
+    "_metadata/verified_contents.json";
+
+std::string GetRootHasheForContent(base::span<const uint8_t> contents,
+                                   size_t block_size) {
+  size_t offset = 0;
+  std::vector<std::string> hashes;
+  // Even when the contents is empty, we want to output at least one hash
+  // block (the hash of the empty string).
+  do {
+    const size_t bytes_to_read = std::min(contents.size() - offset, block_size);
+    std::unique_ptr<crypto::SecureHash> hash(
+        crypto::SecureHash::Create(crypto::SecureHash::SHA256));
+    hash->Update(contents.subspan(offset, bytes_to_read));
+
+    std::string buffer;
+    buffer.resize(crypto::kSHA256Length);
+    hash->Finish(base::as_writable_byte_span(buffer));
+    hashes.push_back(std::move(buffer));
+
+    // If |contents| is empty, then we want to just exit here.
+    if (bytes_to_read == 0) {
+      break;
+    }
+
+    offset += bytes_to_read;
+  } while (offset < contents.size());
+
+  CHECK(block_size % crypto::kSHA256Length == 0);
+  return extensions::ComputeTreeHashRoot(hashes,
+                                         block_size / crypto::kSHA256Length);
+}
+
+// Proxies the file access and checks the verified_contents.json.
+class ComponentContentsAccessorImpl
+    : public ComponentNoChecksContentsAccessorImpl {
+ public:
+  explicit ComponentContentsAccessorImpl(const base::FilePath& component_root)
+      : ComponentNoChecksContentsAccessorImpl(component_root) {
+    verified_contents_ = extensions::VerifiedContents::CreateFromFile(
+        base::make_span(kComponentContentsVerifierPublicKey),
+        component_root.AppendASCII(kVerifiedContentsPath));
+    if (verified_contents_->block_size() % crypto::kSHA256Length != 0) {
```

Comment:
This may occur because it's coming from an external file, so we can't and shouldn't `CHECK` here. Unsupported block size means that somebody's changed the file. The `CHECK` above has been added because `ComputeTreeHashRoot` doesn't check for it, but requires `block_size % sha25len == 0`

Reply:
Acknowledged.